Content.Shared/_EE/Items/OniOnlyComponent.cs

```diff
@@ -0,0 +1,9 @@
+// <summary>
+//   Locks an item to only be pick-upable by onis.
+// </summary>
+[RegisterComponent]
+public sealed partial class OniOnlyComponent : Component
```

Comment:
This really should be given a more generic name, in addition to adding more datafields so as to not hardcode the system. For example, "RestrictedMeleeComponent". We would most likely want to reuse this for other weapons, rather than only use it for Oni equipment. For example, I might feasibly want to make "Biocoded" weapons for Solarians that can only be used by Humans.